### PR TITLE
Document some `outputs` use cases and pitfalls

### PR DESCRIPTION
**Pull request by jaimergp**
PR Checklist:

- [ ] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [X] put any other relevant information below

Comes from discussion at https://github.com/conda-forge/duckdb-feedstock/issues/6#issuecomment-1650705191

### CODE REVIEW

File: src/maintainer/knowledge_base.rst [hunk 1]
```diff
@@ -1309,6 +1309,78 @@ Noarch generic
...
+  
+  - A runtime library and its headers:
+    
+    - `cpp-opentelemetry-sdk <https://github.com/conda-forge/cpp-opentelemetry-sdk-feedstock/blob/main/recipe/meta.yaml>`__
+
+  - A dynamic library and a static version:
+
+    - `libarchive <https://github.com/conda-forge/libarchive-feedstock/blob/main/recipe/meta.yaml>`__
+
+- Distributing the same project with different sets of dependencies. For example:
```
Comment (ocefpaf) [inline src/maintainer/knowledge_base.rst:1334]: This is also a trick to mimic optional dependencies from Python-land. Maybe we should mentioned that here too?

**Review by ocefpaf — APPROVED** [review]
This is awesome @jaimergp! Thanks so much for this PR.

File: src/maintainer/knowledge_base.rst (hunk 1 above)
Comment (jaimergp) [reply src/maintainer/knowledge_base.rst:1334]: Added! 🦸

**Comment by dhirschfeld** [comment]
`typing-extensions` uses `{{ pin_subpackage(name, exact=True) }}`
`importlib-metadta` uses `{{ pin_subpackage('importlib-metadata', max_pin="x.x.x") }}`

but over in https://github.com/conda-forge/duckdb-feedstock/issues/6#issuecomment-1654025349 @isuruf said it should be `{{ version }} *_{{ build_num }}`

Should the preferred version be explicitly documented in the text?

**Comment by dhirschfeld** [comment]
From testing (over in https://github.com/conda-forge/arrow_odbc-feedstock/pull/23) it appears it should be `={{ version }}=*_{{ build_num }}`

**Comment by wshanks** [comment]
A couple suggestions based on my experience in https://github.com/conda-forge/qiskit-terra-feedstock/pull/28:

* If it should be possible to output a Python arch-specific package and a Python noarch package from a single feedstock, document how to do that. I guess in https://github.com/conda-forge/qiskit-terra-feedstock/pull/28 it is almost working other than the cross-compiling case. I wonder if anything weird happens with every arch building the same noarch package? Or maybe the noarch should be set to skip most of the runs?.
* Consider suggesting using `bot: check_solvable: true` as an alternative way to maintain tightly coupled packages with separate feedstocks without `outputs`, especially if the dependent package is simple and could be set to automerge?

**Review by xhochy — APPROVED** [review]

**Pull request merged**